modules/publication/ajax/FileUpload.php

```diff
@@ -0,0 +1,908 @@
+<?php
+/**
+ * Publication file upload & editing handler
+ *
+ * This processes and inserts data for publication uploads & editing
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Publication
+ * @author   David <dblader.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+if (isset($_REQUEST['action'])) {
+    $action = $_REQUEST['action'];
+    if ($action === 'upload') {
+        uploadPublication();
+    } elseif ($action === 'editProject') {
+        editProject();
+    } else {
+        header("HTTP/1.1 400 Bad Request");
+    }
+}
+
+/**
+ * Function for new publications
+ *
+ * @return void
+ */
+function uploadPublication() : void
+{
+    $db   = Database::singleton();
+    $user = \User::singleton();
+    if (!$user->hasPermission('publication_propose')
```

Comment:
As @driusan mentioned elsewhere, you can use `\User::hasAnyPermission()` here if you want.